### DISCUSSION TITLE
Sentry reported an error: Authentication credentials were not provided. (http status: 401)

### DISCUSSION THREAD

**Discussion by briantuju** [vercel/community#583] [Help] [answered]
Hello. I'm stuck trying to solve an error when deploying on **Vercel** (but works on local environment)

![Screenshot from 2022-04-11 11-52-56](https://user-images.githubusercontent.com/69858377/162703689-3495d956-8d53-4449-b2b9-def2860d9981.png)

I followed the [official NextJs Sentry tutorial](https://docs.sentry.io/platforms/javascript/guides/nextjs/) and [this GitHub issue](https://github.com/getsentry/sentry-webpack-plugin/issues/236) but I can't get it to work still.

I have set the environment variable `SENTRY_DSN` and `NEXT_PUBLIC_SENTRY_DSN` on vercel for ALL environments but it won't work.

When I deploy the application to Vercel using the **vercel cli** using `vercel --prod`, it works.
What could I be doing wrong?

Kindly help as this is really dragging me down. Below are my configuration for Sentry and NextJs

![Screenshot from 2022-04-11 12-17-44](https://user-images.githubusercontent.com/69858377/162707503-3fd2b15e-a34f-4b2a-8ad5-72719d8ad92e.png)
![Screenshot from 2022-04-11 12-18-00](https://user-images.githubusercontent.com/69858377/162707533-4780a42c-8b28-4557-8df7-c0589ddc8caa.png)
![Screenshot from 2022-04-11 12-18-47](https://user-images.githubusercontent.com/69858377/162707547-273dd1d1-f048-4e1e-b243-ee7d90094d46.png)

**Comment by bhishp** [comment, marked as answer]
You mention that you have set `SENTRY_DSN` and `NEXT_PUBLIC_SENTRY_DSN` but have you set  `SENTRY_AUTH_TOKEN`? The error in your screenshot mentions  "Authentication credentials were not provided" so it may be that the token is not present?

## Sentry/Vercel Integration

I was also having this same authentication issue on preview builds with Vercel, and I used the nextjs wizard + [Sentry Vercel integration](https://docs.sentry.io/product/integrations/deployment/vercel/) to generate everything for me. However, it seems that when the Vercel integration adds environment variables, it adds them for **prod only**, this means when `next build` is ran on a Preview build, it tries to call `sentry release`, but there is no auth token. By enabling the environment variable in Preview too I was able to build successfully.

You could argue that releases don't need to be generated for previews, so you could conditionally disable the release with something like `process.env.VERCEL_ENV !== "production"`

**Reply by briantuju** [reply]
Thank you for your input. It was really helpful.

**Reply by jamesmills** [reply]
Hit the same issue. 

Tried to add a condition to only include Sentry when in production but something is still trying to run. We (@jonspark) tried a couple of ways.

```
...
Sentry CLI Plugin: Command failed: /vercel/path0/node_modules/@sentry/cli/sentry-cli releases new be087184c71959954a6838b7b668249684857c79
--
16:28:52.555 | error: API request failed
16:28:52.555 | caused by: sentry reported an error: Authentication credentials were not provided. (http status: 401)
...
```

Don't really need/want feature branch builds to trigger a release in Sentry but the only way I've been able to get around this is to update `SENTRY_ORG`, `SENTRY_PROJECT` and `SENTRY_AUTH_TOKEN` to be available to Production and Preview.

**Reply by bhishp** [reply]
I did it this way @jamesmills...

```js
// in next.config.js
module.exports = withSentryConfig(moduleExports, {
  dryRun: process.env.VERCEL_ENV !== "production"
});
```

Assumes you're using nextjs and the sentryWebpackPlugin (which is the default configuration with the wizard right now). Not sure what else `dryRun: true` prevents but it definitely stops the release from happening

**Comment by briantuju** [comment]
For whoever who is experiencing the same issue, I was able to solve this by using the **Sentry Integration on Vercel**  and updating the `SENTRY_AUTH_TOKEN` to both preview and production.

Here is a [helpful link](https://github.com/getsentry/sentry-javascript/issues/4383)

**Reply by fernandops26** [reply]
This worked for me too deploying on aws amplify host. Thanks!

**Comment by lobsterkatie** [comment]
UPDATE: A variant of @bhishp's approach is now baked in: https://github.com/getsentry/sentry-javascript/pull/5603

(IOW, we no longer try to run the CLI in preview or dev environments on vercel.)

**Comment by ipavlyk-startupsoft** [comment]
npm script:

`   
    "build:dev": "NODE_ENV=dev next build; next export",
`

next.config.js

`
const sentryWebpackPluginOptions = {
  silent: true,
  dryRun: process.env.NODE_ENV !== "production",
}
`

**dryRun** will disable release fty (including Authentication) and make it possible to build locally using npm script above

**Reply by lobsterkatie** [reply]
> `const sentryWebpackPluginOptions = { silent: true, dryRun: process.env.NODE_ENV !== "production", }`

This should not be necessary, since we now don't run the webpack plugin at all if not in a prod environment. See https://github.com/getsentry/sentry-javascript/pull/5603 which I linked above.

**Reply by stephent** [reply]
@lobsterkatie it seems that it may be necessary - I just installed the Sentry/Vercel integration and it attempts a release on a preview build. (All dependencies in the project are up-to-date, `"@sentry/nextjs": "7.27.0"`). Here's what I see on pushing to a preview branch:

```
Failed to compile.
--
22:31:40.197 |  
22:31:40.197 | Sentry CLI Plugin: Command failed: /vercel/path0/node_modules/@sentry/cli/sentry-cli releases new Ig1Yp13Ow7mw0MkujFVtA
22:31:40.197 | error: API request failed
22:31:40.197 | caused by: sentry reported an error: Authentication credentials were not provided. (http status: 401)
```

Is some other configuration required to preview the release running in a Vercel preview build?

**Comment by jainChetanUnolo** [comment]
My build was failing on the server with error above but  it was running successfully on local, all I did then remove .sentryclirc from gitignore and it worked. I don't why that is the case as this file should not be commited in fact sentry by default puts it into .gitignore file, so can anyone explain please?

**Reply by lobsterkatie** [reply]
You need to set the auth token as an environment variable on your server: https://docs.sentry.io/platforms/javascript/guides/nextjs/manual-setup/#use-environment-variables.

(I've filed [an issue](https://github.com/getsentry/sentry-javascript/issues/6123) in our backlog to improve the error message here.)

**Reply by amyegan** [reply]
Thank you, Katie!

**Reply by alishaevn** [reply]
I did not want to commit the .sentryclirc file since it has an auth token in it. I also wasn't using the vercel and sentry integration.

adding the `SENTRY_AUTH_TOKEN` to my environment variables list on vercel for production only and redeploying my `main` branch solved the problem for me.
